DocumentDB/main.tf

```diff
@@ -0,0 +1,70 @@
+/* 
+* # DocumentDB 
+*/
+
+# Terraform code to create an Amazon DocumentDB 
+
+# Create the DocumentDB parameter group
+resource "aws_docdb_cluster_parameter_group" "this" {
+  count       = var.enable ? 1 : 0
+  name        = "parameter-group-${var.database_name}"
+  description = "DB cluster parameter group."
+  family      = var.cluster_family
+
+  dynamic "parameter" {
+    for_each = var.parameters
+    content {
+      apply_method = lookup(parameter.value, "apply_method", null)
+      name         = parameter.value.name
+      value        = parameter.value.value
+    }
+  }
+
+  tags = local.common_tags
+}
+
+# Create the DocumentDB cluster
+resource "aws_docdb_cluster" "this" {
+  count                           = var.enable ? 1 : 0
+  cluster_identifier              = var.database_name
+  master_username                 = var.master_username
+  master_password                 = var.master_password
+  backup_retention_period         = var.backup_retention_period
+  preferred_backup_window         = var.backup_window
+  final_snapshot_identifier       = lower(var.database_name)
+  apply_immediately               = var.apply_immediately
+  deletion_protection             = var.deletion_protection
+  storage_encrypted               = var.storage_encrypted
+  kms_key_id                      = var.kms_key_id
+  snapshot_identifier             = var.snapshot_identifier
+  vpc_security_group_ids          = var.vpc_security_group_ids
+  db_subnet_group_name            = aws_docdb_subnet_group.this[0].name
+  db_cluster_parameter_group_name = aws_docdb_cluster_parameter_group.this[0].name
+  engine                          = var.engine
+  engine_version                  = var.engine_version
+
+  tags = local.common_tags
+
+  depends_on = [aws_docdb_cluster_parameter_group.this]
+}
+
+# DocumentDB instance
+resource "aws_docdb_cluster_instance" "this" {
+  count              = var.enable ? var.cluster_size : 0
```

Comment:
Would we ever want to create a DocDB cluster with no instances?  (again, grain of salt with this question because my total lack of experience with the resource type 😅 )

Reply:
Ah, I think I see what you're going for - the `var.enable` is a way for you to delete/recreate the cluster without deleting the TF?  If that's the case, please ignore my question!

Reply:
Yes, you are completely right - basically this determined if the resource should be created or deleted without deleting the TF